server/mem_agent/engine.py

```
@@ -0,0 +1,334 @@
+        "allow_installs": allow_installs,
+        "allowed_path": allowed_path,
+        "blacklist": blacklist or [],
+        "available_functions": available_functions or {},
+        "log": log,
+    }
+
+    env = os.environ.copy()
+    env["SANDBOX_PARAMS"] = base64.b64encode(pickle.dumps(params)).decode()
+
+    try:
+        result = subprocess.run(
```

Ruff (0.14.1)

`subprocess` call: check for execution of untrusted input

(S603)